### PR TITLE
Expose more simbody methods needed by advanced users

### PR DESCRIPTION
**Pull request by aymanhab**
Fixes issue #3414, #3490

### Brief summary of changes
Added 3 methods previously not accessible to advanced users.

### Testing I've completed
Diffed resulting Model.java files and the three methods were added and are utilizing other existing classes

### Looking for feedback on...
Documentation/description of the methods and if there're other related questions/concerns

### CHANGELOG.md (choose one)
- updated.

- - -
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3618)

### CODE REVIEW

**Comment by aymanhab** [comment]
@nickbianco not sure if you have a minute to review this. Thank you

File: OpenSim/Simulation/Model/Model.h [hunk 1]
```diff
@@ -522,7 +522,21 @@ OpenSim_OBJECT_NONTEMPLATE_DEFS(Model, ModelComponent);
     GeneralForceSubsystem allocated by this %Model. **/
     SimTK::GeneralForceSubsystem& updForceSubsystem() 
     {   return *_forceSubsystem; }
-
+    /** (Advanced) Get read only access to internal Simbody RigidBodyForces at Dynamics stage **/
+    const SimTK::Vector_<SimTK::SpatialVec>& getRigidBodyForces(const SimTK::State& state)
```
Comment (nickbianco) [inline OpenSim/Simulation/Model/Model.h:526]: Does `SimTK::Vector_<SimTK::SpatialVec>` get wrapped properly in the bindings? I don't see a `%template` for it.

**Review by nickbianco — COMMENTED** [review]
One quick question.

File: OpenSim/Simulation/Model/Model.h (hunk 1 above)
Comment (aymanhab) [reply OpenSim/Simulation/Model/Model.h:526]: Yes it does here's the signature from the newly generated Model.java
public VectorOfSpatialVec getGravityBodyForces(State state) {
  ....
}

**Comment by aymanhab** [comment]
Failure is the other intermittent osx failure unrelated to this PR.

File: OpenSim/Simulation/Model/Model.h (hunk 1 above)
Comment (nickbianco) [reply OpenSim/Simulation/Model/Model.h:526]: Should we create our own SWIG template to follow the convention for other vectors like `VectorVec3`? Perhaps `VectorSpatialVec`? I definitely wouldn't know to search for the signature `VectorOfSpatialVec` without knowing it existed already. (I'm assuming `VectorOfSpatialVec` is auto-generated by SWIG).

**Comment by aymanhab** [comment]
@nickbianco The name of the class is not swig generated, it's defined here  
https://github.com/opensim-org/opensim-core/blob/4b75cbc2d7007f9ee47c7e82c9f1da3e34027648/Bindings/simbody.i#L220
since at least 2015

I agree there's inconsistency with other naming conventions but it's hard to tell why it was done this way or who is using these classes so a rename would be breaking backward compatibility. We can discuss further at dev meeting

**Comment by nickbianco** [comment]
@aymanhab gotcha, not sure how I missed that while searching. In that case, it doesn't make sense to potentially break backwards-compatibility with this PR.

**Review by nickbianco — APPROVED** [review]
LGTM.

**Comment by aymanhab** [comment]
Thanks @nickbianco Much appreciated 👍

**Pull request merged**